### DISCUSSION TITLE
Welcome to MantaRay Discussions!

### DISCUSSION THREAD

**Discussion by Sonderwoods** [Sonderwoods/MantaRay#91] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

And do bring all of your radiance questions!

Best, Mathias

**Comment by lmk2021** [comment]
Hi Mathias
Thank you very much for sharing you great work
I am a lighting designer working primarily on daylighting. My setup is to model in Rhino and to work with Radiance on a linux machine. So MantaRay could fit into my workflow just perfectly.
I did some tests and I think I am able to use the basic functions. There is one issue I would like to bring up.
My RAYPATH on the linux machine looks like .:/home/martin/mylib:/usr/local/lib/ray. After connect with MantaRay and without changing anything it is .:/usr/local/radiance/lib. Where does this come from? Do you set the RAYPATH somewhere? If I connect to my linux box with putty I get the right RAYPATH.
Would be grate if you can help me with that.
Thank you very much and greetings from Austria
Martin

**Comment by Sonderwoods** [comment]
Hi Martin,

It's a bit confusing but SSH works in two ways (I'm by no means an expert
on that). Apparently when you ssh with putty you end up using your host
computers bash settings from .profile and .bashrc (and one more which I
forgot).

However theres also a way to send SSH commands without loading the bash
profile. That is (unfortunately) how mantaray works as this is the only
method I could find in the ssh.net framework which I am using.

Workaround? Input a text panel to the Connect component with your own
prefixes. Such as `export path=....`

This command will set your path but only for that one mini ssh session
(which is only one command!). The prefix command is automatically added
before any of your execute_ssh commands (right click on execute_ssh
component and disable prefixes if you want to disable the feature).

My "prefixes" are set as default if you don't input anything. If you add a
panel to the prefix output of the Ssh_connect component you should be able
to see the prefixes.. you can copy that text and add to a text panel if you
with to overwrite them. The prefix is saved here in the code btw.
https://github.com/Sonderwoods/MantaRay/blob/main/MantaRay/Helpers/SSH_Helper.cs#L61

Let me know if this solves your issue!

Best, M

fre. 4. nov. 2022 15.28 skrev lmk2021 ***@***.***>:

**Comment by lmk2021** [comment]
Hi Mathias
Thank you very much for your answer. It´s quite clear now.
My workaround was to add the "export path=" to every ExecuteSSH manualy. I was not aware of the prefix way.
My problem at the moment is, that I do not know where to connect the text panel with the prefix to the connect component. Do I have an old version?
![grafik](https://user-images.githubusercontent.com/82712879/200111552-c4d4a8dc-d651-4724-9b9c-975374f8e0f9.png)

Best regards
Martin

**Comment by Sonderwoods** [comment]
Hey my bad on the input, maybe it's not released yet.

Are you able to build it yourself? In that case try building the dev branch.

If not, please be patient untill I get some time to build it, probably end
of next week. Just spent my quota of "computer time" this weekend on the
LINK Dashboards plugin release on food4rhino. :D

I'll do a food4rhino release for this one soonish also. Just finalizing the
examples.

lør. 5. nov. 2022 09.46 skrev lmk2021 ***@***.***>:

**Comment by lmk2021** [comment]
Hi Mathias
Thank you so much for your time and I really do not want to be annoying
I build the dev tree and now the prefix input is there and works as explained - great
Then I had the idea to put the cd to the working dir also into the prefix. For me that is cd <LinuxHome>.
That gives an error like (sorry for the German language)
![grafik](https://user-images.githubusercontent.com/82712879/200166356-a6ecaab7-8870-4952-b0e6-fa2d536bc58b.png)
Please understand this just as an observation. I am glad to test how to integrate your great work into my workflow.
Best regards
Martin

**Reply by Sonderwoods** [reply]
I just build it so it's available to download for non-geeks :-) 

Looks like the ssh command is not happy for the `;`  .. You might be able to pass a newline instead. I my workflows i havent CD'ed to the paths, i just set a global like  `<workfolder> = "/my_work_folder/project/etc...."` with the global overrides component    and always use  `rtrace <workfolder>/model.oct` in the execute commands
